File: 백준/Silver/10845. 큐/큐.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n; cin >> n;
    queue<int> q;
    while(n--){
        string cmd; cin >> cmd;
        if(cmd == "push"){
            int x; cin >> x;
            q.push(x);
        }
        else if(cmd == "pop"){
            if(q.size()){
                cout << q.front() << '\n'; q.pop();
            }
            else cout << -1 << '\n';
        }
        else if(cmd == "size") cout << q.size() << '\n';
        else if(cmd == "empty") cout << (q.size() ? 0 : 1) << '\n';
        else if(cmd == "front") cout << (q.size() ? q.front() : -1) << '\n';
        else if(cmd == "back") cout << (q.size() ? q.back() : -1) << '\n';
    }
    return 0;
}

File: 백준/Silver/10845. 큐/큐.py
import sys
def print(*args, sep=" ", end="\n"): return sys.stdout.write(sep.join(map(str, args)) + end)
def input(): return sys.stdin.readline().rstrip()

from collections import deque

def main():
    n = int(input())
    dq = deque()
    ans = []
    for _ in range(n):
        cmd = input().split()
        if cmd[0] == 'push': dq.append(int(cmd[1]))
        elif cmd[0] == 'pop': ans.append(dq.popleft() if dq else -1)
        elif cmd[0] == 'size': ans.append(len(dq))
        elif cmd[0] == 'empty': ans.append(0 if dq else 1)
        elif cmd[0] == 'front': ans.append(dq[0] if dq else -1)
        elif cmd[0] == 'back': ans.append(dq[-1] if dq else -1)
    print('\n'.join(map(str, ans))+'\n')
    return 0
if __name__ == '__main__':
    sys.exit(main())